pkg/envoy_access_log_server/service_v3.go

```diff
@@ -0,0 +1,225 @@
+		if sa := p.DownstreamLocalAddress.GetSocketAddress(); sa != nil {
+			m["downstreamLocalAddress"] = sa.GetAddress()
+			m["downstreamLocalPort"] = fmt.Sprint(sa.GetPortValue())
+		}
+	}
+	m["routeName"] = p.RouteName
+	if ts := p.StartTime; ts != nil {
+		if t, err := ptypes.Timestamp(ts); err == nil {
+			m["startTime"] = t.Format(time.RFC3339)
+		}
+	}
+	if p.TimeToFirstDownstreamTxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToFirstDownstreamTxByte); err == nil {
+			m["timeToFirstDownstreamTxByte"] = duration
+		}
+	}
+	if p.TimeToFirstUpstreamRxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToFirstUpstreamRxByte); err == nil {
+			m["timeToFirstUpstreamRxByte"] = duration
+		}
+	}
+	if p.TimeToFirstUpstreamTxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToFirstUpstreamTxByte); err == nil {
+			m["timeToFirstUpstreamTxByte"] = duration
+		}
+	}
+	if p.TimeToLastDownstreamTxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToLastDownstreamTxByte); err == nil {
+			m["timeToLastDownstreamTxByte"] = duration
+		}
+	}
+	if p.TimeToLastRxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToLastRxByte); err == nil {
+			m["timeToLastRxByte"] = duration
+		}
+	}
+	if p.TimeToLastUpstreamRxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToLastUpstreamRxByte); err == nil {
+			m["timeToLastUpstreamRxByte"] = duration
+		}
+	}
+	if p.TimeToLastUpstreamTxByte != nil {
+		if duration, err := pbDurationDeterministicString(p.TimeToLastUpstreamTxByte); err == nil {
+			m["timeToLastUpstreamTxByte"] = duration
+		}
+	}
+	m["upstreamCluster"] = p.UpstreamCluster
+	if p.UpstreamLocalAddress != nil {
+		if sa := p.UpstreamLocalAddress.GetSocketAddress(); sa != nil {
+			m["upstreamLocalAddress"] = sa.GetAddress()
+			m["upstreamLocalPort"] = fmt.Sprint(sa.GetPortValue())
+		}
+	}
+	if p.UpstreamRemoteAddress != nil {
+		if sa := p.UpstreamRemoteAddress.GetSocketAddress(); sa != nil {
+			m["upstreamRemoteAddress"] = sa.GetAddress()
+			m["upstreamRemotePort"] = fmt.Sprint(sa.GetPortValue())
+		}
+	}
+	m["upstreamTransportFailureReason"] = p.UpstreamTransportFailureReason
+	m["sampleRate"] = strconv.FormatFloat(p.SampleRate, 'f', -1, 64)
+
+	return m
+}
+
+type envoyV3AccessLogEntryHttpRequestProperties envoy_data_accesslog_v3.HTTPRequestProperties
+
+func (request envoyV3AccessLogEntryHttpRequestProperties) StringMap() stringmap.StringMap {
+	result := stringmap.StringMap{}
+
+	result["authority"] = request.Authority
+	result["forwardedFor"] = request.ForwardedFor
+	result["originalPath"] = request.OriginalPath
+	result["path"] = request.Path
+	result["referer"] = request.Referer
+
+	// request headers are encoded with `http_` prefix (to keep the scheme used by Nginx)
+	if request.RequestHeaders != nil {
+		for header := range request.RequestHeaders {
+			result["http_"+header] = request.RequestHeaders[header]
```

Comment:
Cannot help myself, my 2 cents: Stating anything about Nginx here does not make any sense to me at all. There is absolutely no relation. If all the keys are named the same as the Envoy protobuf, I'd stick to it even here using `requestHeaders_<headerName>` same with `responseHeaders_<headerName>` and `responseTrailers_<TrailerName>`.

Reply:
I agree.

But as was discussed offline, this was one of the @lksv (product owner) requirements - accompanied with an assumption that most of the users of this module would be familiar with Nginx much more than with Envoy. I don't know if that holds true, but it is what it is. :)